### ISSUE TITLE
local config

### ISSUE THREAD

**Issue by iferres** [replikation/poreCov#76]
Hi, thanks for implementing this workflow. New fan here!
Just a minor request for the `local` profile. There're some processes which one would expect to require small resources (e.g. nanoplot), but the `local` profile assigns `cpus = params.cores`. In my case, I'm running it on a local server with 32 cpus. That leads to "allocating" the whole server to a single instance of (say) nanoplot, when this process only require little more than 1 cpu at most, slowing down the whole wf execution (blocks parallel execution of multiple instances).
Very easy to solve with a new configuration, my request is just to improve the experience to users new to nextflow. I think for most of these processes giving a `cpus = 2` as default would be enough, and would allow multiple parallel instances. Some of them may require a little more tho, as `artic` and `bwa`.

**Comment by hoelzer** [comment]
Hi @iferres, thanks for the message and for using the wf!

Yes, you are right and I agree that for some processes the default CPU usage can be reduced to allow more parallelization.

@replikation I think something like:

```
process {
    withLabel:  artic       { cpus = params.cores }
    withLabel:  bwa         { cpus = params.cores }
    withLabel:  fastcov     { cpus = 4 }
    withLabel:  ggplot2     { cpus = 4 }
    withLabel:  guppy_cpu   { cpus = params.cores }
    withLabel:  guppy_gpu   { cpus = params.cores }
    withLabel:  kraken2     { cpus = params.cores }
    withLabel:  krona       { cpus = 1 }
    withLabel:  minimap2    { cpus = params.cores }    
    withLabel:  nanoplot    { cpus = 2 }
    withLabel:  nextclade   { cpus = params.cores }
    withLabel:  pangolin    { cpus = 4 }
    withLabel:  president   { cpus = 1 }
    withLabel:  pycoqc      { cpus = 4 }
    withLabel:  snp_sites   { cpus = 4 }
    withLabel:  ubuntu      { cpus = 1 } 
}
```

I am also not sure if `guppy_gpu` needs `params.cores` bc/ it anyway runs on the GPU so does not need to block CPUs? (however, it's anyway the first step all other processes have to wait for)

And, @iferres if you are not aware of it: there is also an additional parameter `--max_cores` for the `local` profile. By that, you can configure how many CPUs are block in total on your machine. So lets say you have a local machine w/ 32 CPUs but you only want to use 24 in total, you could specify:

```
--cores 8 --max_cores 24
```

so e.g. `bwa` would in maximum run 3 times in parallel.

**Comment by replikation** [comment]
Hi @iferres glad you like it :)

yes, we still have #58 open, so feedback here is of course welcome :) . Hardcoding nanoplot down to 4 makes e.g. sense I think. @hoelzer the cpus are for the demultiplexing part after the calling which is slow with only a few threads.

Usually, multithreading is a bit more difficult on one machine (e.g. we want to avoid overloading) so the best way (in my experience) is using the --cores to adjust the level of "parallel" runs as @hoelzer suggested.

**Comment by replikation** [comment]
added to f91964118cf42d2f07a520de5a6c87b66f6e9ac5

**Comment by iferres** [comment]
Thank you! And sorry for my late response. I said nanoplot as an example. I'd probably set other defaults as well, but I'd need to benchmark them before. I ran it on 12 barcodes with `cores = 4` and it was pretty fast.
Nice to have the `--max_cores` parameter, I may or may not copy this implementation from your wf to mine :P .
Bests!

**Comment by replikation** [comment]
@iferres yep let me know 👍🏻
